stan/math/rev/mat/fun/matrix_exp_action_handler.hpp

```diff
@@ -0,0 +1,117 @@
+#ifndef STAN_MATH_REV_MAT_FUN_MATRIX_EXP_ACTION_HANDLER_HPP
+#define STAN_MATH_REV_MAT_FUN_MATRIX_EXP_ACTION_HANDLER_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/rev/core.hpp>
+#include <vector>
+
+double l1norm(const Eigen::MatrixXd& m) {
+  return m.colwise().lpNorm<1>().maxCoeff();
+}
+
+namespace stan {
+namespace math {
+
+/*
+ * The implemention of the work by Awad H. Al-Mohy and Nicholas J. Higham
+ * "Computing the Action of the Matrix Exponential,
+ * with an Application to Exponential Integrators"
+ * Read More: https://epubs.siam.org/doi/abs/10.1137/100788860
+ *
+ * Calculates exp(mat*t)*b, where mat & b are matrices,
+ * and t is double.
+ *
+ */
+class matrix_exp_action_handler {
+  static constexpr int p_max = 8;
+  static constexpr int m_max = 55;
+  static constexpr double tol = 1.1e-16;
+  static const std::vector<double> theta_m_single_precision;
+  static const std::vector<double> theta_m_double_precision;
+
+ public:
+  matrix_exp_action_handler() {}
+
+  /* Perform the matrix exponential action exp(A*t)*B
+   * @param [in] mat matrix A
+   * @param [in] b matrix B
+   * @param [in] t double t, e.g. time.
+   * @return matrix exp(A*t)*B
+   */
+  inline Eigen::MatrixXd action(const Eigen::MatrixXd& mat,
+                                const Eigen::MatrixXd& b,
+                                const double& t = 1.0) {
+    Eigen::MatrixXd A = mat;
+    double mu = A.trace() / A.rows();
+    for (int i = 0; i < A.rows(); ++i) {
+      A(i, i) -= mu;
+    }
+
+    int m{0}, s{0};
```

Comment:
Is there a reason to use brace initializers rather than just `m = 0`?

Reply:
Likely not. Must be some left-over from a prototype using `auto`.